src/services.js

```diff
@@ -472,8 +479,22 @@ export class Services {
    * @return {!Promise<?Object<string,(string|Array<string>)>>}
    */
   static geoForOrNull(win) {
-    return (/** @type {!Promise<?Object<string,(string|Array<string>)>>} */
-      (getElementServiceIfAvailable(win, 'geo', 'amp-geo', true)));
+    /**
+     * If there is a stubbed amp-geo extension wait for it to load before trying
+     * to get the service.  Prevents a race condition when everything but
+     * amp-geo is in cache.  If there is no stub then it's either loaded,
+     * not present, or the 'geo' service was defiend by a test. In those cases
+     * we don't wait around for an extension that may not exist.
+     */
+    if (stubbedElements.includes['amp-geo']) {
```

Comment:
Please instead move this code into `getElementServiceIfAvailable` so that all callers are fixed.